tic_tac_toe/game.py

```diff
@@ -1,104 +1,145 @@
+    enumerate(board)]]
+    
+    all_combinations = across_combinations + vertical_combinations + \
+    diagonal_2_combination + diagonal_1_combination
+
+    for lists in all_combinations:
+        test_for_equality = []
+        for pos in lists:
+            test_for_equality.append(board[pos[0]][pos[1]])
+        if len(set(test_for_equality)) == 1 and test_for_equality[0] != '-':
+            return test_for_equality[0]
+        else:
+            continue
+    else:
+        return None
+
+            
 
 # public interface
 def start_new_game(player1, player2):
-    """
-    Creates and returns a new game configuration.
-    """
-    pass
-
+    game = {
+        'player1': player1,
+        'player2': player2,
+        'board': [
+            ["-", "-", "-"],
+            ["-", "-", "-"],
+            ["-", "-", "-"],
+        ],
+        'next_turn': "X",
+        'winner': None
+        }
+
+    return game
 
 def get_winner(game):
-    """
-    Returns the winner player if any, or None otherwise.
-    """
-    pass
+    return game["winner"]
 
 
 def move(game, player, position):
-    """
-    Performs a player movement in the game. Must ensure all the pre requisites
-    checks before the actual movement is done.
-    After registering the movement it must check if the game is over.
-    """
-    pass
+    winner = _check_winning_combinations(game["board"], player)
+    if _board_is_full(game["board"]) == True and game["winner"] is not None:
+        raise InvalidMovement('Game is over.')
+    
+    if not _position_is_valid(position):
+        raise InvalidMovement('Position out of range.')
+    
+    next_turn = get_next_turn(game)
+    if next_turn is None:
+        raise InvalidMovement("Game is over.")
+    
+    if not _position_is_empty_in_board(position, game["board"]):
+        raise InvalidMovement('Position already taken.')
+        
+    if next_turn != player:
+        raise InvalidMovement('"{}" moves next'.format(next_turn))
+    
+    game['board'][position[0]][position[1]] = player
+    
+    winner = _check_winning_combinations(game["board"], player)
+    if _board_is_full(game["board"]) and winner is None:
+        raise GameOver('Game is tied!')
+    elif winner is not None:
+        game['winner'] = winner
+        raise GameOver('"{}" wins!'.format(winner))
+    else:
+        if game["next_turn"] == game['player1']:
+            game['next_turn'] = game['player2']
+        else:
+            game['next_turn'] = game['player1']
+    
+    
+
+
 
 
 def get_board_as_string(game):
-    """
-    Returns a string representation of the game board in the current state.
-    """
-    pass
+    return '''
+{}  |  {}  |  {}
+--------------
+{}  |  {}  |  {}
+--------------
+{}  |  {}  |  {}
+'''.format(game["board"][0][0], game["board"][0][1], \
+    game["board"][0][2], game["board"][1][0], game["board"][1][1], \
+    game["board"][1][2], game["board"][2][0], game["board"][2][1], \
+    game["board"][2][2])
+    
```

Comment:
Nice idea for creating a string representation of the board. It makes what you are doing very clear!